### ISSUE TITLE
Updates to Design/Dev Intake Form (Apps Form)

### ISSUE THREAD

**Issue by acshields** [appirio-tech/connect-app#4095]
**Updates to this form:** https://connect.topcoder.com/new-project/app

### Update 1
Update “Briefly describe your goals” to read:
Question: “What are the main business objectives you are trying to achieve?”

### Update 2
**_Conditional Dependencies:_**
User has selected:
- “Design”
- “Development & QA”
- “Design AND Development & QA”
- “Design AND Development & QA AND Deployment” 
- On the “Where should your app work?” question, if the user selects “Web Browser.”

**_Updates Required_**
Include a new question/answer.
- Question: “Which browsers need to be supported?”
- Answer: [Checklist Format]
-Chrome
-Safari
-Firefox
-Edge
-Internet Explorer
-No Preferences

### Update 3
**_Conditional Dependencies:_**
User has selected:
- “Development & QA,” or 
- “Design AND Development & QA” 
-"Design AND Development & QA AND Deployment"

**_Updates Required:_**
After the “Technology Stack” question section, the user will have this optional question to complete:
- Question: “Are there any technology dependencies?”
- Question Sub-Text: If yes, please provide a brief description.
- Answer: Checklist + Text Field, if item is checked
-APIs
-Backend
-Database

### Update 4
**_Conditional Dependencies:_**
User has selected:
- “Design”
- “Development & QA”
- “Design AND Development & QA” 
- “Design AND Development & QA AND Deployment”

**_Updates Required:_**
Essentially, we need to pull the “add-on” section of this form, which is currently hidden, into the main form’s question/answer section. We will not be calculating the cost/timeline implications of these items, so if you can lift and land the question/answer copy easily that would be great. Here is what should be included:
- Optional Question: Screen Features
- Answers: [Checklist]
-Registration: Register and login with email/password creation, with edit password or recover password options.
-Enterprise Login: Integration with an existing authorization/authentication mechanism.
-Email Login: Login with an email/password.
-Social Login: Register and login with a social media third party system. 
-Account Settings: Adjust account settings or indicate preferences, such as notification cadence.
-Profiles: Create and edit user profiles.
-Invitations: Invite others to use the application.
-Introductions: Inform users on core application functionality in introductory screens.
-Onboarding: Virtually walk users through the application.
-Dashboard: A centralized dashboard for the application.
-Search: Supported search capabilities within the application.
-Location-Based Services: Identification of user geographic location.
-Camera (Audio & Video): Camera-enabled audio/video in the application.
-File Upload: Upload of documents, links, pictures, video, etc.
-Notifications: In-platform and/or emailed notifications.
-Tagging: Tagging of users or attributes in the application.
-Help/FAQs: Help/FAQ dedicated content.
-Marketplace: Allow users to buy, sell, or rent products.
-Product Listings: Present product listings, with dedicated descriptive pages for each product.
-Shopping Cart: Allow users to save items before purchasing.
-Payments: Accept payments in the application.
-Ratings & Reviews: Enable users to rate/review people, products, or services.
-Activity Feed: Show users a feed of recent application activity.
-Messaging: Enable direct communication between 2+ users.
-Contact Us: Enable user to contact administrator(s).
-Admin Panel: An administrative panel for direct user management, content, or the application.
-Reporting: Report and/or extract data from the application.

### Update 5
**_Conditional Dependencies:_**
- User has selected:
- “Development & QA”
- “Design AND Development & QA” 
- “Design AND Development & QA AND Deployment” 

**_Updates Required:_**
**- Optional Question: Technology Requirements**
- Answers: [Checklist]
-Social Media Integration: Integration to social media providers.
-SSO Integration: Integration to enterprise single sign-on capability.
-Third Party System Integration: Integration with 3rd party systems with the capability to retrieve or post data.
-SMS Gateway Integration: Integration with external SMS gateway/provider for SMS notifications.
-Containerized Code: Docker containerized codebase for easier deployment and maintenance.
-Continuous Integration/Continuous Deployment: Established CI/CD pipeline.
-Error Logging: Logging of all errors, exceptions, warnings, and debug information.
-Analytics Implementation: Track user behavior and application usage.
-Email (SMTP Server) Set-up: Development and configuration of SMTP server to send email notifications. 

**- Optional Question: Testing Requirements**
- Answers: [Checklist]
-Unit Tests: Application unit tests to support code coverage.
-Unstructured Testing: Real-world testing completed without test scripts to identify UI/UX or usability issues.
-UAT/Beta Test Cycle: Enable your users to test and provide feedback.
-UAT/Beta Test Implementations: Implementation of changes provided through user feedback.
-Structured Testing: Creation and/or execution of test cases focused on functional requirements and cross-browser device coverage.
-App Certification: Certify the application against CPU/battery/memory usage, inactive/active/low battery behavior, performance under outages.
-Performance Testing: Test the robustness, availability, and reliability of the application under defined business scenarios or concurrent users.
-Performance Tuning: Identify performance issues and actionable items for improvement.

- **Question Section: Security Requirements**
- Question 1: Will this application contain or transmit confidential information, sensitive financial data, or personally identifiable information (PII)?
- Question 1 Answer: Yes/No Toggle Button
- Question 2: Select all required security information.
- Question 2 Answers: [Checklist]
-Vulnerability Scanning: Scan this application for vulnerabilities.
-User Action Audit: Record specific user actions.

### Update 6
**_Conditional Dependencies:_**
User has selected:
- “Development & QA”
- “Design AND Development & QA” 
- “Design AND Development & QA AND Deployment”

**_Update Required:_**
- Question: Will you supply test data or will Topcoder create it?
- Answers: [Radio Buttons]
-We will provide obfuscated data.
-Topcoder will create data.

### Update 7
**_Conditional Dependencies:_**
- User has selected “Design,” “Development & QA,” or “Design AND Development & QA,” or "Design AND Development & QA AND Deployment" 
- All previous questions, either currently in the form or listed above, have been asked.

**_Update Required:_**
- Question: Described this application’s ideal workflow.
- Answer: [Free Text Field - High or no character limit]

### Update 8
**_Conditional Dependencies:_**
- User has selected “Design,” “Development & QA,” or “Design AND Development & QA,” "Design AND Development & QA AND Deployment" 
- All previous questions, either currently in the form or listed above, have been asked.

**_Update Required:_**
- Optional Question 1: User Descriptions
-Field 1 - Standard User: [Free Text- High or no character limit]
-Field 2 - Admin User: [Free Text - High or no character limit]
-Field 3 - Super Admin User: [Free Text - - High or no character limit]
- Optional Question 2: How many users do you intend to support?
- Answer: [Free Text]

### Update 9
**_Conditional Dependencies:_**
User has selected:
- “Development & QA”
- “Design AND Development & QA” 
- "Development & QA AND Deployment"
- “Design AND Development & QA AND Deployment”

**_Update Required:_**
This is a required question that should immediately be asked after the user has answered "What do you need?" question and has provided one of the conditional dependency combinations above.
- Field Label: "What outcome do you expect?"
- Answer Options [Radio Buttons]
-"Proof of Concept"
Include this sub-text for the Proof of Concept answer option: "Build a limited application to prove a concept is workable. Screens will be limited and code is not production-ready."
-"Minimum Viable Product"
Include this sub-text for the Minimum Viable Product answer option: "Code is production-ready but with a limited set of screens and functionality."
-"Production-Ready"
Include this sub-text for the Production-Ready answer option: "All requested screens and functionality is fully developed, tested, and ready for production deployment."

**Comment by acshields** [comment]
@RishiRajSahu - Rash requested one additional question to be included in this form, which is now reflected in Update 9.

**Comment by RishiRajSahu** [comment]
@acshields 
Updates 1-3 are done.
Updates 4-8, I need to know where these questions should be placed ? I mean before/after which existing questions ?
Update 9 - "What do you need?" is a checkbox type question, so currently as soon as we select any checkbox(es) the other dependent question pop-up on the form, so this new question about outcome, do we want to show it after the dependent questions or should it be present after those dependent questions ?

**Comment by acshields** [comment]
@RishiRajSahu - Thank you for the update!

Questions 4-8 can be placed in their current sequence above, after the new questions 1-3 and all current questions in this screenshot have been asked: 
<img width="470" alt="Screen Shot 2020-09-09 at 8 22 57 AM" src="https://user-images.githubusercontent.com/41118149/92604329-05777900-f276-11ea-8c07-bc9000564757.png">

For Question 9, this can be displayed after the new question 8 (so at the end of the form). In terms of the Q&A for Question 9, this will just be a simple checklist answer. No triggered text fields or additional (dependent) questions need to be generated for this question. In the sub-text descriptions, I was just hoping we could provide the sub-text description similar to how we do in other checklist answers (we can keep the current function where the sub-text displays after the user clicks the answer). Here is a similar example of how it should look:
<img width="479" alt="Screen Shot 2020-09-09 at 8 29 53 AM" src="https://user-images.githubusercontent.com/41118149/92605058-e0cfd100-f276-11ea-80ef-172da1a5108c.png">

**Comment by acshields** [comment]
@RishiRajSahu - I will post my screen recording in Slack, but as I tested out the version of this form in dev I ran into some issues. I am not sure what was triggering the issued I faced (might be several factors), but hopefully the video will help.

@lakshmiathreya - as a preface, this form (Apps intake, or the design/dev form) could benefit from a QA tester's support (if codejam or deepak have capacity). Essentially, this form has a lot of conditional logic built into it that dictates when questions should be asked, or not, and it's important to ensure that we are asking all required questions based upon the conditional logic.
